lib/response-handler.js

```diff
@@ -110,25 +113,27 @@ module.exports = (bluebird, error, logger, parseLinkHeader, ShelfError, URI) =>
         links = [];
         resolvedLinkHeaders = [];
 
-        if (response.headers.link) {
+        if (response && response.headers && response.headers.link) {
             logger.debug(`Got link attribute: ${response.headers.link}`);
             links = parseLinkHeader(response.headers.link);
             logger.debug(`Parsed link header: ${JSON.stringify(links, null, 4)}`);
 
             // If we received just one link header item, push it into an array.
             if (!Array.isArray(links.item)) {
-                links = [links.item];
+                links.item = [links.item];
             }
 
             logger.debug(`Parsed out links: ${JSON.stringify(links, null, 4)}`);
         }
 
-        links.item.forEach((item) => {
-            var resolvedLink;
+        if (links.item) {
```

Comment:
Link is set to an empty array on line 113. On line 116 is an if statement that will set links.item if it's entered into. If it's not, links remains an empty array, and trying to iterate over it with forEach on line 130 will throw a TypeError.